ufl/form.py

```diff
@@ -69,7 +69,159 @@ def _sorted_integrals(integrals):
     return tuple(all_integrals)  # integrals_dict
 
 
-class Form(object):
+class BaseForm(object):
+    """Description of an object containing arguments"""
+
+    # Slots is kept empty to enable multiple inheritance with other classes.
+    __slots__ = ()
+    _ufl_is_abstract_ = True
+    _ufl_required_methods_ = ('_analyze_form_arguments', "ufl_domains")
+
+    def __init__(self):
+        # Internal variables for caching form argument data
+        self._arguments = None
+
+    # --- Accessor interface ---
+    def arguments(self):
+        "Return all ``Argument`` objects found in form."
+        if self._arguments is None:
+            self._analyze_form_arguments()
+        return self._arguments
+
+    # --- Operator implementations ---
+
+    def __eq__(self, other):
```

Comment:
Then, `__eq__` should be removed in `Form` since they are identical.

Reply:
done :)